### ISSUE TITLE
Overfitting

### ISSUE THREAD

**Issue by YingcaiDong** [YingcaiDong/cnn-text-classification-tf#1]
After running the original [code](https://github.com/dennybritz/cnn-text-classification-tf), I could not reproduce the same result mentioned in the [article](http://www.wildml.com/2015/12/implementing-a-cnn-for-text-classification-in-tensorflow/).
<img width="1025" src="https://user-images.githubusercontent.com/22902303/32382551-f6854dce-c08b-11e7-8237-4a6ef3b6455b.png">
Loss result

As the training data's loss getting lower and lower, the validation data is getting bigger. So I implement **early stop** during the training process. The user is able to adjust one parameter to tune the early stop process.
```
# Early stop parameters
tf.flags.DEFINE_integer("patience", 3, "max epoch allowed after delta rises")
```
`FLAGS.patience` number of epochs with no improvement after which training will be stopped.

After implementing the early stop method, the training process is able to stop before the neuron network gets overfitting. While saves training time.
<img width="975" alt="screen shot 2017-11-03 at 12 22 56 pm" src="https://user-images.githubusercontent.com/22902303/32384656-fb08a048-c091-11e7-865b-7ee1b7b7c29d.png">
Loss result, training stops at step = 1800, loss = 0.191899

<img width="1000" alt="screen shot 2017-11-03 at 12 22 35 pm" src="https://user-images.githubusercontent.com/22902303/32384660-ff7dfa38-c091-11e7-8802-bf57fcb609e3.png">
traing accuracy  = 0.966667

checkpoint save at step 1300, where the best loss was found, where training accuracy = 0.890625, training loss = 0.317399. Evaluation loss = 0.589816, acc = 0.727955.
